### ISSUE TITLE
bug?

### ISSUE THREAD

**Issue by jwijffels** [quanteda/spacyr#92]
Hello,

I'm trying to compare Spacy and UDPipe (https://CRAN.R-project.org/package=udpipe) regarding a number of things namely 

1. Easy of use
2. Annotation possibilities
3. Annotation languages
4. Annotation speed
5. Annotation accuracy of the models

While I was comparing the model accuracies (code at https://github.com/jwijffels/udpipe-spacy-comparison) I've encountered a bug in the spacyr package 
I've tried to make a reproducible example from this which is shown below:

```
library(spacyr)
spacy_initialize(model = "en", python_executable = "C:/Users/Jan/Anaconda3/python.exe")
txt <- c("I -- which I try to do, as best as I possibly can.", 
         "I -- which I try to do, as best as I possibly can.", 
         "I -- which I try to do, as best as I possibly can.")
x <- lapply(txt, FUN=function(x){
  names(x) <- x
  spacy_parse(txt, pos = TRUE, tag = TRUE, lemma = TRUE, dependency = TRUE, entity = FALSE) 
})
x[[3]]
   doc_id sentence_id token_id    token    lemma   pos tag head_token_id dep_rel
1   text1           1        1        I   -PRON-  PRON PRP             1    ROOT
2   text1           1        2       --       -- PUNCT   :             1   punct
3   text1           1        3    which    which   ADJ WDT             7    dobj
4   text1           1        4        I   -PRON-  PRON PRP             5   nsubj
5   text1           1        5      try      try  VERB VBP             1   relcl
6   text1           1        6       to       to  PART  TO             7     aux
7   text1           1        7       do       do  VERB  VB             5   xcomp
8   text1           1        8        ,        , PUNCT   ,             5   punct
9   text1           1        9       as       as   ADV  RB            10  advmod
10  text1           1       10     best     best   ADV RBS             5  advmod
11  text1           1       11       as       as   ADP  IN            14    mark
12  text1           1       12        I   -PRON-  PRON PRP            14   nsubj
13  text1           1       13 possibly possibly   ADV  RB            14  advmod
14  text1           1       14      can      can  VERB  MD            10   advcl
15  text1           1       15        .        . PUNCT   .             1   punct
16  text2           1        1        I   -PRON-  PRON PRP             1    ROOT
17  text2           1        2       --       -- PUNCT   :             1   punct
18  text2           1        3    which    which   ADJ WDT             7    dobj
19  text2           1        4        I   -PRON-  PRON PRP             5   nsubj
20  text2           1        5      try      try  VERB VBP             1   relcl
21  text2           1        6       to       to  PART  TO             7     aux
22  text2           1        7       do       do  VERB  VB             5   xcomp
23  text2           1        8        ,        , PUNCT   ,             5   punct
24  text2           1        9       as       as   ADV  RB            10  advmod
25  text2           1       10     best     best   ADV RBS             5  advmod
26  text2           1       11       as       as   ADP  IN            14    mark
27  text2           1       12        I   -PRON-  PRON PRP            14   nsubj
28  text2           1       13 possibly possibly   ADV  RB            14  advmod
29  text2           1       14      can      can  VERB  MD            10   advcl
30  text2           1       15        .        . PUNCT   .             1   punct
31  text3           1        1        I   -PRON-  PRON PRP             1    ROOT
32  text3           1        2       --       -- PUNCT   :             1   punct
33  text3           1        3    which    which   ADJ WDT             7    dobj
34  text3           1        4        I   -PRON-  PRON PRP             5   nsubj
35  text3           1        5      try      try  VERB VBP             1   relcl
36  text3           1        6       to       to  PART  TO             7     aux
37  text3           1        7       do       do  VERB  VB             5   xcomp
38  text3           1        8        ,        , PUNCT   ,             5   punct
39  text3           1        9       as       as   ADV  RB            10  advmod
40  text3           1       10     best     best   ADV RBS             5  advmod
41  text3           1       11       as       as   ADP  IN            14    mark
42  text3           1       12        I   -PRON-  PRON PRP            14   nsubj
43  text3           1       13 possibly possibly   ADV  RB            14  advmod
44  text3           1       14      can      can  VERB  MD            10   advcl
45  text3           1       15        .        . PUNCT   .             1   punct
spacy_finalize()
```

**Comment by kbenoit** [comment]
I think the bug here is in the code of the `lapply()` function (and the fact that R's scoping rules will look in the parent environment for an object undefined in the function environment!).  The argument to `spacy_parse()` should be `x` but in your example, it's `txt`, hence each element returned has parsed the full, parsed input vector of length three.

```r
x <- lapply(txt, FUN=function(x){
    spacy_parse(x, pos = TRUE, tag = TRUE, lemma = TRUE, dependency = TRUE, entity = FALSE) 
})
x[[3]]
# 1   text1           1        1        I   -PRON-  PRON PRP             1        ROOT
# 2   text1           1        2       --       -- PUNCT   :             1       punct
# 3   text1           1        3    which    which   ADJ WDT             5 dobj||xcomp
# 4   text1           1        4        I   -PRON-  PRON PRP             5       nsubj
# 5   text1           1        5      try      try  VERB VBP             1       relcl
# 6   text1           1        6       to       to  PART  TO             7         aux
# 7   text1           1        7       do       do  VERB  VB             5       xcomp
# 8   text1           1        8        ,        , PUNCT   ,             5       punct
# 9   text1           1        9       as       as   ADV  RB            10      advmod
# 10  text1           1       10     best     best   ADV  RB             1      advmod
# 11  text1           1       11       as       as   ADP  IN            14        mark
# 12  text1           1       12        I   -PRON-  PRON PRP            14       nsubj
# 13  text1           1       13 possibly possibly   ADV  RB            14      advmod
# 14  text1           1       14      can      can  VERB  MD            10       advcl
# 15  text1           1       15        .        . PUNCT   .             1       punct
```

same results as
```r
lapply(txt, spacy_parse, pos = TRUE, tag = TRUE, lemma = TRUE, dependency = TRUE, entity = FALSE)[[3]]
```
if you prefer simpler syntax.

Also note that it's not necessary to `lapply()` over documents. In this case it misleadingly labels the third list element document as `text1` because it assigns  "text*n*" as a document name if the input element is unnamed, so in effect in lapplying it, you end up with three "text1"s. Better just to input the character vector to `spacy_parse()` as is.
```r

**Comment by jwijffels** [comment]
My excuses you're right on this. I was trying to reproduce with a simple example of what I saw appearing when I ran a whole bunch of annotations yesterday but I've made an error. 

The problem that I saw appearing yesterday is that in the data.frame of annotations the token_id within a sentence suddenly started to increase and included sometimes annotations from a previous sentence. This only happened when there were 2 times the same sentences which had to be annotated. Let me see if I can reproduce this.
